### ISSUE TITLE
Fatal Exception: Could not write to keychain

### ISSUE THREAD

**Issue by tankista** [bugfender/BugfenderSDK-iOS#34]
bugfender version 1.5.4

I am getting some crashes, but only 10 users so far out of thousands. I have linked Security.framework but I am not following any entitlements.plist. Using CocoaPods. This happens only on latest version on my app, perhaps I am doing something wrong, but I have not touched much of the code. Can you please help me to address this issue?

`+[BFKeychain load:]`
_Keychain can not be accessed, probably Security.framework is not linked or entitlements.plist file is missing_

```
Fatal Exception: Could not write to keychain
0  CoreFoundation                 0x184baed8c __exceptionPreprocess
1  libobjc.A.dylib                0x183d685ec objc_exception_throw
2  BugfenderSDK                   0x101d7fec0 +[BFKeychain load:] (BFKeychain.m:38)
3  BugfenderSDK                   0x101daf818 +[BFSystemUtils bugfenderDeviceIdentifier] (BFSystemUtils.m:303)
4  BugfenderSDK                   0x101d7e608 __28+[Bugfender activateLogger:]_block_invoke (BugfenderSDK.m:55)
5  libdispatch.dylib              0x1844a0a60 _dispatch_client_callout
6  libdispatch.dylib              0x1844a4268 dispatch_once_f$VARIANT$mp
7  BugfenderSDK                   0x101d7e5a4 +[Bugfender activateLogger:] (BugfenderSDK.m:58)
8  Trak                           0x100c040f0 -[AppDelegate application:didFinishLaunchingWithOptions:] (AppDelegate.m:138)
9  UIKit                          0x18e7badbc -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:]
10 UIKit                          0x18e7ba1c4 -[UIApplication _callInitializationDelegatesForMainScene:transitionContext:]
11 UIKit                          0x18e7875e0 -[UIApplication _runWithMainScene:transitionContext:completion:]
12 UIKit                          0x18edb7b1c __111-[__UICanvasLifecycleMonitor_Compatability _scheduleFirstCommitForScene:transition:firstActivation:completion:]_block_invoke
13 UIKit                          0x18e786dd0 +[_UICanvas _enqueuePostSettingUpdateTransactionBlock:]
14 UIKit                          0x18e786c6c -[__UICanvasLifecycleMonitor_Compatability _scheduleFirstCommitForScene:transition:firstActivation:completion:]
15 UIKit                          0x18e785afc -[__UICanvasLifecycleMonitor_Compatability activateEventsOnly:withContext:completion:]
16 UIKit                          0x18f41b84c __82-[_UIApplicationCanvas _transitionLifecycleStateWithTransitionContext:completion:]_block_invoke
17 UIKit                          0x18e7851ec -[_UIApplicationCanvas _transitionLifecycleStateWithTransitionContext:completion:]
18 UIKit                          0x18f200ac8 __125-[_UICanvasLifecycleSettingsDiffAction performActionsForCanvas:withUpdatedScene:settingsDiff:fromSettings:transitionContext:]_block_invoke
19 UIKit                          0x18f34ebf8 _performActionsWithDelayForTransitionContext
20 UIKit                          0x18e784c0c -[_UICanvasLifecycleSettingsDiffAction performActionsForCanvas:withUpdatedScene:settingsDiff:fromSettings:transitionContext:]
21 UIKit                          0x18e7845a8 -[_UICanvas scene:didUpdateWithDiff:transitionContext:completion:]
22 UIKit                          0x18e7815e0 -[UIApplication workspace:didCreateScene:withTransitionContext:completion:]
23 UIKit                          0x18e781330 -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:]
24 FrontBoardServices             0x1873af470 -[FBSSceneImpl _didCreateWithTransitionContext:completion:]
25 FrontBoardServices             0x1873b7d6c __56-[FBSWorkspace client:handleCreateScene:withCompletion:]_block_invoke_2
26 libdispatch.dylib              0x1844a0a60 _dispatch_client_callout
27 libdispatch.dylib              0x1844a8170 _dispatch_block_invoke_direct$VARIANT$mp
28 FrontBoardServices             0x1873e3878 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__
29 FrontBoardServices             0x1873e351c -[FBSSerialQueue _performNext]
30 FrontBoardServices             0x1873e3ab8 -[FBSSerialQueue _performNextFromRunLoopSource]
31 CoreFoundation                 0x184b57404 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
32 CoreFoundation                 0x184b56c2c __CFRunLoopDoSources0
33 CoreFoundation                 0x184b5479c __CFRunLoopRun
34 CoreFoundation                 0x184a74da8 CFRunLoopRunSpecific
35 GraphicsServices               0x186a59020 GSEventRunModal
36 UIKit                          0x18ea91758 UIApplicationMain
37 Trak                           0x100c114b4 main (main.m:16)
38 libdyld.dylib                  0x184505fc0 start
```

**Comment by jgimenez** [comment]
Hi @tankista , thanks for the report. We're aware of this intermittent issue and we are working on a solution, we expect to issue an updated version of the framework this or next week.

**Comment by tankista** [comment]
@jgimenez hey, any progress on this issue? thanks

**Comment by jgimenez** [comment]
Fixed in [release 1.5.5](https://github.com/bugfender/BugfenderSDK-iOS/releases/tag/1.5.5), please let us know if you're still having any issues after upgrading
